app/assets/javascripts/discourse/controllers/forgot-password.js.es6

```diff
@@ -52,6 +52,42 @@ export default Ember.Controller.extend(ModalFunctionality, {
       });
 
       return false;
+    },
+
+    emailLogin() {
+      if (this.get('submitDisabled')) return false;
+
+      this.set('disabled', true);
+
+      ajax('/u/email-login', {
+        data: { login: this.get('accountEmailOrUsername').trim() },
```

Comment:
Resetting password process goes down the same path, it sends the very same token to user.  I am not sure the process. The dialog may need some polish here?